### DISCUSSION TITLE
Pylsp disable plugins properly

### DISCUSSION THREAD

**Discussion by vakker** [joaotavora/eglot#925] [Q&A]
Hi,

I'm using pylsp and I'm trying to disable some plugins, but they are still active in the buffers.
E.g. (based on [this](https://github.com/joaotavora/eglot/discussions/709)) I have in my init.el:
```
(after! eglot
  :init
  (setq-default eglot-workspace-configuration
                '((pylsp
                   (plugins
                    (pycodestyle (enabled . nil))
                    (pyflakes (enabled . t))
                    (flake8 (enabled . nil))
                    ))))
  (add-hook 'eglot-managed-mode-hook (lambda () (eldoc-mode -1)))
  )
```
I'm using the Doom `after!` mechanism.
Pyflakes works as expected, but I'm still getting all the Pycodestyle messages.

Is this how it's supposed to be used?

**Comment by joaotavora** [comment]
No idea how after! works, but I think you might be looking for `eglot-ignored-server-capabilities`. Look up its docstring.

**Reply by vakker** [reply]
I was doing some debugging and now it seems to work (the snippet that I added above). Maybe I had a typo or I just didn't save it properly, or something similarly stupid. Thanks for your reply anyway.

Just a weird observation, in the variable help it shows up as:
```lisp
((pylsp
  (plugins
   (pycodestyle
    (enabled . t))
   (pyflakes
    (enabled . t))
   (flake8
    (enabled)))))
```
Is that correct for the `nil` values?

**Reply by joaotavora** [reply]
no idea, I haven't ever used `pylsp` I think.  But `(foo . nil)` in lisp is the very same as `(foo)`
